include/sys/spa.h

```diff
@@ -219,7 +220,7 @@ typedef struct zio_cksum_salt {
  * G		gang block indicator
  * B		byteorder (endianness)
  * D		dedup
- * X		encryption (on version 30, which is not supported)
+ * X		encryption
```

Comment:
I think it should be sufficient to describe it.  Maybe in this (existing) section, around ```
 * checksum[4]	256-bit checksum of the data this bp describes
```
you could say that if the X bit is set, checksum[2-3] is used for the MAC.  X-bit BP's that are encrypted use a different layout described below, but authenticated and indirect BP's use the above layout but the last 2 words of the checksum are actually the MAC.